docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,59 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow all
+kind of customisations since it would make the system even more complex to test
+and maintain over time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
+depending on the architecture of the platform or depending on the different
+use cases and applications running on the top of it.
+
+It can be:
+
+- The BootstrapConfig,
+- The various roles and taints we set on the node objects of the cluster
+- The labels and annotations we would set on all K8s objects
```

Comment:
This is too much free form, and contradicts the sentence `We also don't want to allow all kind of customisations`. We cannot allow users to add whichever labels/annotations on any K8s object **we** deploy - it may break the deployment itself. I also don't see why they would like to do so on our K8s objects. Whatever isn't deployed by MetalK8s is not in scope for this feature.

Reply:
Okay so I remove the part about labels and annotations but if we don't want to allow the administrator setting such labels or annotation, maybe we should make sure he cannot do so on our K8s objects?

Reply:
Making sure they can't do stuff we don't document would mostly be managed through RBAC. Though I don't see how we can prevent a cluster admin from doing whatever he needs/wants.